tempfile and py3 incompatibilities

Bunch of str/bytes issues:

[ERROR] Traceback (most recent call last):
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/site-packages/turbinia-20181004-py3.5.egg/turbinia/workers/__init__.py", line 607, in run_wrapper
    self.result = self.run(evidence, self.result)
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/site-packages/turbinia-20181004-py3.5.egg/turbinia/workers/analysis/jenkins.py", line 85,
 in run
    (report, priority, summary) = self.analyze_jenkins(version, credentials)
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/site-packages/turbinia-20181004-py3.5.egg/turbinia/workers/analysis/jenkins.py", line 165
, in analyze_jenkins
    weak_passwords = bruteforce_password_hashes(credentials_registry.keys())
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/site-packages/turbinia-20181004-py3.5.egg/turbinia/lib/utils.py", line 124, in bruteforce
_password_hashes
    fh.write('\n'.join(password_hashes))
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/tempfile.py", line 622, in func_wrapper
    return func(*args, **kwargs)
TypeError: a bytes-like object is required, not 'str'

[ERROR] JenkinsAnalysisTask Task failed with exception: [a bytes-like object is required, not 'str']
[INFO] Traceback (most recent call last):
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/site-packages/turbinia-20181004-py3.5.egg/turbinia/workers/__init__.py", line 607, in run_wrapper
    self.result = self.run(evidence, self.result)
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/site-packages/turbinia-20181004-py3.5.egg/turbinia/workers/analysis/jenkins.py", line 85,
 in run
    (report, priority, summary) = self.analyze_jenkins(version, credentials)
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/site-packages/turbinia-20181004-py3.5.egg/turbinia/workers/analysis/jenkins.py", line 165, in analyze_jenkins
    weak_passwords = bruteforce_password_hashes(credentials_registry.keys())
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/site-packages/turbinia-20181004-py3.5.egg/turbinia/lib/utils.py", line 124, in bruteforce
_password_hashes
    fh.write('\n'.join(password_hashes))
  File "/home/romaing/venvs/turbinia-dev3/lib/python3.5/tempfile.py", line 622, in func_wrapper
    return func(*args, **kwargs)
TypeError: a bytes-like object is required, not 'str'

--- a/turbinia/lib/utils.py
+++ b/turbinia/lib/utils.py
@@ -117,7 +117,7 @@ def bruteforce_password_hashes(password_hashes, timeout=300):
     TurbiniaException if execution failed.
   """
 
-  with tempfile.NamedTemporaryFile(delete=False) as fh:
+  with tempfile.NamedTemporaryFile(delete=False, mode='w+') as fh:
     password_hashes_file_path = fh.name
     fh.write('\n'.join(password_hashes))
 

--- a/turbinia/processors/google_cloud.py
+++ b/turbinia/processors/google_cloud.py
@@ -59,7 +59,7 @@ def GetLocalInstanceName():
       'http://metadata.google.internal/computeMetadata/v1/instance/name', None,
       {'Metadata-Flavor': 'Google'})
   try:
-    instance = urllib.request.urlopen(req).read()
+    instance = urllib.request.urlopen(req).read().decode('utf-8')
   except urllib.error.HTTPError as e:
     raise TurbiniaException('Could not get instance name: {0!s}'.format(e))
 

--- a/turbinia/workers/analysis/jenkins.py
+++ b/turbinia/workers/analysis/jenkins.py
@@ -88,7 +88,7 @@ class JenkinsAnalysisTask(TurbiniaTask):
     result.report_priority = priority
 
     # Write the report to the output file.
-    with open(output_file_path, 'w') as fh:
+    with open(output_file_path, 'wb') as fh:
       fh.write(output_evidence.text_data.encode('utf8'))
       fh.write('\n'.encode('utf8'))
 

--- a/turbinia/workers/analysis/wordpress.py
+++ b/turbinia/workers/analysis/wordpress.py
@@ -72,7 +72,7 @@ class WordpressAccessLogAnalysisTask(TurbiniaTask):
     result.report_priority = priority
 
     # Write the report to the output file.
-    with open(output_file_path, 'w') as fh:
+    with open(output_file_path, 'wb') as fh:
       fh.write(output_evidence.text_data.encode('utf-8'))
 
     # Add the resulting evidence to the result object.

--- a/turbinia/workers/hadoop.py
+++ b/turbinia/workers/hadoop.py
@@ -133,7 +133,7 @@ class HadoopAnalysisTask(TurbiniaTask):
       result.report_data = output_evidence.text_data
 
       # Write the report to the output file.
-      with open(output_file_path, 'w') as fh:
+      with open(output_file_path, 'wb') as fh:
         fh.write(output_evidence.text_data.encode('utf8'))
         fh.write('\n'.encode('utf8'))
 

